src/Components/Game.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from "react";
+import Board from "./Board";
+import { GameProps, isNotNull } from "../Types/TypeProps";
+import GameInfo from './GameInfo';
+/**
+ * Handle Game logic
+ * @param props 
+ * @returns 
+ */
+const Game = (props:GameProps) => {
+  /**
+   * Game's state declarations
+   */
+  const [history, setHistory] = useState(props.historyVal);
+  const [stepNumber, setStepNumber] = useState(props.stepNumberVal);
+  const [xIsNext, setXisNext] = useState(props.xIsNextVal);
+  const [counter, setCounter] = useState(history.length);
+
+  /**
+   * Handle each board's cell action when clicked
+   * @param {*} i
+   */
+  const handleClick = (i: number) => {
+    const gameHistory = history.slice(0, stepNumber + 1);
+    const current = gameHistory[gameHistory.length - 1];
+    const squares = current.squares.slice();
+    
+
+    // Prevent further cell clicks after winning
+    if (calculateWinner(squares) || squares[i]) return;
+
+    // Determine next player
+    squares[i] = xIsNext ? "X" : "O";
+
+    // Update states
+    setHistory(gameHistory.concat([{ squares: squares }]));
+    setCounter(history.length);
+    setStepNumber(gameHistory.length);
+    setXisNext((prevXisNext) => !prevXisNext);
+  };
+
+  /**
+   * Reset game
+   */
+  const reset = () => {
+    setHistory(props.historyVal);
+    setCounter(1);
+    setStepNumber(props.stepNumberVal);
+    setXisNext(props.xIsNextVal);
+  };
+
+  /**
+   * Go to previous Move
+   */
+  const prevMove = () => {
+    if (stepNumber === 0) return;
+    setStepNumber((prevStep) => prevStep - 1);
+    setXisNext((next) => !next);
+  };
+
+  /**
+   * Go to next move
+   */
+  const nextMove = () => {
+    if (stepNumber >= counter || counter === 1) return;
+    setStepNumber((prevStep) => prevStep + 1);
+    setXisNext((next) => !next);
+  };
+
+  /**
+   * Update time travel steps with step number
+   * @param {*} step
+   */
+  const jumpTo = (step: number) => {
+    setStepNumber(step);
+    setXisNext(step % 2 === 0);
+  };
+
+    /**
+   * Calculate game winner
+   * @param {*} squares
+   */
+  const calculateWinner = (squares: Array<string>) => {
+    const lines: Array<Array<number>> = [
+      [0, 1, 2],
+      [3, 4, 5],
+      [6, 7, 8],
+      [0, 3, 6],
+      [1, 4, 7],
+      [2, 5, 8],
+      [0, 4, 8],
+      [2, 4, 6]
+    ];
+    for (let i = 0; i < lines.length; i++) {
+      const [a, b, c] = lines[i];
+      if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
+        return {winner: squares[a], position: [a,b,c]} ;
```

Comment:
Everywhere that this `position` variable is referenced I would change it to `winningSquares` or something
```suggestion
        return {winner: squares[a], winningSquares: [a,b,c]} ;
```